File: api/app/Models/CalibrationLog.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class CalibrationLog extends Model
{
    protected $fillable = [
        'isotank_id',
        'item_name',
        'description',
        'planned_date',
        'vendor',
        'status',
        'calibration_date',
        'valid_until',
        'serial_number',
        'replacement_serial',
        'replacement_calibration_date',
        'notes',
        'created_by',
        'performed_by',
    ];

    protected $casts = [
        'planned_date' => 'date',
        'calibration_date' => 'date',
        'valid_until' => 'date',
        'replacement_calibration_date' => 'date',
        'status' => 'string',
    ];

    public function isotank(): BelongsTo
    {
        return $this->belongsTo(MasterIsotank::class, 'isotank_id');
    }

    public function creator(): BelongsTo
    {
        return $this->belongsTo(User::class, 'created_by');
    }

    public function performer(): BelongsTo
    {
        return $this->belongsTo(User::class, 'performed_by');
    }
}

File: mobile/lib/data/models/calibration_log.dart
class CalibrationLog {
  final int id;
  final int isotankId;
  final String itemName;
  final String? serialNumber;
  final String? description;
  final DateTime? plannedDate;
  final String? vendor;
  final String status;
  final String? isotankNumber;

  CalibrationLog({
    required this.id,
    required this.isotankId,
    required this.itemName,
    this.serialNumber,
    this.description,
    this.plannedDate,
    this.vendor,
    required this.status,
    this.isotankNumber,
  });

  factory CalibrationLog.fromJson(Map<String, dynamic> json) {
    return CalibrationLog(
      id: json['id'],
      isotankId: json['isotank_id'],
      itemName: json['item_name'],
      serialNumber: json['serial_number'],
      description: json['description'],
      plannedDate: json['planned_date'] != null ? DateTime.parse(json['planned_date']) : null,
      vendor: json['vendor'],
      status: json['status'],
      isotankNumber: json['isotank'] != null ? json['isotank']['iso_number'] : null,
    );
  }
}
